gluten-core/src/main/scala/io/glutenproject/execution/SortMergeJoinExecTransformer.scala

```diff
@@ -239,10 +225,150 @@ case class SortMergeJoinExecTransformer(
 
   override def getChild: SparkPlan = streamedPlan
 
-  override def doValidate(): Boolean = false
+  def genJoinParametersBuilder(): com.google.protobuf.Any.Builder = {
+    val (isBHJ, isNullAwareAntiJoin, buildHashTableId) = genJoinParameters()
+    // Start with "JoinParameters:"
+    val joinParametersStr = new StringBuffer("JoinParameters:")
+    // isBHJ: 0 for SHJ, 1 for BHJ
+    // isNullAwareAntiJoin: 0 for false, 1 for true
+    // buildHashTableId: the unique id for the hash table of build plan
+    joinParametersStr.append("isBHJ=").append(isBHJ).append("\n")
+      .append("isNullAwareAntiJoin=").append(isNullAwareAntiJoin).append("\n")
+      .append("buildHashTableId=").append(buildHashTableId).append("\n")
+      .append("isExistenceJoin=").append(
```

Comment:
Looks some parameters are not needed for SMJ, such as `isBHJ`, `isNullAwareAntiJoin`, `buildHashTableId`. Also, if we are using `isBHJ` to illustrate a BHJ, can we do the similar thing for SMJ before Substrait supportes physical plan?

Reply:
Added isSMJ to illustrate a SMJ.